### DISCUSSION TITLE
defaultConnector for TreeLayoutView

### DISCUSSION THREAD

**Discussion by adalekin** [clientIO/joint#1925] [Q&A] [answered]
Hi,

Is there any way to change defaultConnector for TreeLayoutView?

Thanks.

**Comment by kumilingus** [comment, marked as answer]
The [defaultConnector](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.Paper.prototype.options.defaultConnector) can be set on the `Paper` or each individual link.

---

Sometimes the `TreeLayoutView` needs to create a new link (when the user drags one of the roots and attaches it to another branch). In this case, the `TreeLayoutView` will look at the paper option [defaultLink](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.Paper.prototype.options.defaultLink).

If you want to change the shape of the links completely, you can use [updateVertices](https://resources.jointjs.com/docs/rappid/v3.6/layout.TreeLayout.html) `TreeLayout` model option (e.g. set `updateVertices` to `null` and the `defaultConnector` to  [curve](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#connectors.curve)).

**Reply by adalekin** [reply]
Yeah, I get it. Probably I didn't describe the situation clear enough. Please see the screenshot. I would like to change the shape of a preview connector of `TreeLayoutView `.
<img width="1583" alt="Снимок экрана 2022-12-05 в 12 04 38" src="https://user-images.githubusercontent.com/164028/205597697-e60d4965-afce-4fde-9647-a95d3efc1a77.png">

**Reply by kumilingus** [reply]
Currently, it is necessary to override the `updateConnectionPreview` method.

```js
const TreeLayoutView = joint.ui.TreeLayoutView.extend({

    updateConnectionPreview: function(source, target, vertices) {
        this.svgPreviewConnection.attr({
            // generate a curve here instead
            d: `
                M ${source.x} ${source.y}
                ${vertices.map(v => `L ${v.x} ${v.y}`).join(' ')}
                L ${target.x} ${target.y}
            `
        });
    },

});
```
